sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,325 @@
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      // Force any not signed MSFAA to be signed to ensure that new
+      // ones created will be the only ones available to be updated.
+      await db.msfaaNumber.update(
+        { dateSigned: IsNull() },
+        { dateSigned: getISODateOnlyString(new Date()) },
+      );
+      // Cancel any pending MSFAA.
+      await db.msfaaNumber.update(
+        { cancelledDate: IsNull() },
+        { cancelledDate: getISODateOnlyString(new Date()) },
+      );
+    });
+
+    it("Should process an MSFAA response with confirmations and a cancellation and update all records when the file is received as expected.", async () => {
+      // Arrange
+      const msfaaInputData = [
+        MSFAA_PART_TIME_MARRIED,
+        MSFAA_PART_TIME_OTHER_COUNTRY,
+        MSFAA_PART_TIME_RELATIONSHIP_OTHER,
+      ];
+      const createdMSFAARecords = await saveMSFAATestInputsData(
+        db,
+        msfaaInputData,
+      );
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      mockDownloadFiles(sftpClientMock, [
+        MSFAA_PART_TIME_RECEIVE_FILE_WITH_CANCELATION_RECORD,
+      ]);
+
+      // Act
+      const processResult = await processor.processMSFAAResponses(job);
+
+      // Assert
+      expect(processResult).toStrictEqual([
+        {
+          processSummary: [
+            `Processing file ${MSFAA_PART_TIME_RECEIVE_FILE_WITH_CANCELATION_RECORD}.`,
+            "File contains:",
+            "Confirmed MSFAA records (type R): 2.",
+            "Cancelled MSFAA records (type C): 1.",
+            "Record from line 1, updated as confirmed.",
+            "Record from line 3, updated as confirmed.",
+            "Record from line 2, updated as canceled.",
+          ],
+          errorsSummary: [],
+        },
+      ]);
+      // Assert that the file was deleted from SFTP.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+      // Find the updated MSFAA records previously created.
+      const msfaaIDs = createdMSFAARecords.map((msfaa) => msfaa.id);
+      const msfaaUpdatedRecords = await db.msfaaNumber.find({
+        select: {
+          msfaaNumber: true,
+          dateSigned: true,
+          serviceProviderReceivedDate: true,
+          cancelledDate: true,
+          newIssuingProvince: true,
+        },
+        where: {
+          id: In(msfaaIDs),
+        },
+        order: {
+          msfaaNumber: "ASC",
+        },
+      });
+      expect(msfaaUpdatedRecords).toHaveLength(msfaaInputData.length);
+      const [fistSignedMSFAA, cancelledMSFAA, secondSignedMSFAA] =
+        msfaaUpdatedRecords;
+      // Validate fist confirmed record.
+      expect(fistSignedMSFAA.dateSigned).toBe("2021-11-20");
+      expect(fistSignedMSFAA.serviceProviderReceivedDate).toBe("2021-11-21");
+      // Validate cancelled record.
+      expect(cancelledMSFAA.cancelledDate).toBe("2021-11-24");
+      expect(cancelledMSFAA.newIssuingProvince).toBe("ON");
```

Comment:
just to check `serviceProviderReceivedDate` is not for canceled entry

Reply:
Right, `serviceProviderReceivedDate`  is set while confirming an MSFAA.